elfi/methods/bo/acquisition.py

```diff
@@ -550,6 +551,188 @@ def _evaluate_logpdf(theta):
+
+    L_{1:t}(\theta^*) \approx 2 * \sum_{i=1}^s (\omega^i * p^2(\theta^i)
+                                * w_{1:t+1})(theta^i, \theta^*), where
+
+    \omega^i is an importance weight,
+    p^2(\theta^i) is the prior squared, and
+    w_{1:t+1})(theta^i, \theta^*) is the expected variance of the unnormalised ABC posterior
+    at \theta^i after running the simulation model with parameter \theta^*
+
+    References
+    ----------
+    [1] arXiv:1704.00520 (Järvenpää et al., 2017)
+
+    """
+
+    def __init__(self, quantile_eps=.01, integration='grid', d_grid=.2,
+                 n_samples_imp=100, iter_imp=2, sampler='nuts', n_samples=2000,
+                 sigma_proposals_metropolis=None, *args, **opts):
+        """Initialise ExpIntVar.
+
+        Parameters
+        ----------
+        quantile_eps : int, optional
+            Quantile of the observed discrepancies used in setting the discrepancy threshold.
+        integration : str, optional
+            Integration method.
+            (Options: importance, grid.)
+        d_grid : float, optional
+            Grid tightness.
+        n_samples_imp : int, optional
+            Number of importance samples.
+        iter_imp : int, optional
+            Gap between acquisition iterations in performing importance sampling.
+        sampler : string, optional
+            Sampler for generating random numbers from the proposal distribution for IS.
+            (Options: metropolis, nuts.)
+        n_samples : int, optional
+            Chain length for the sampler that generates the random numbers
+            from the proposal distribution for IS.
+        sigma_proposals_metropolis : array_like, optional
+            Standard deviation proposals for tuning the metropolis sampler.
+
+        """
+        super(ExpIntVar, self).__init__(quantile_eps, *args, **opts)
+        self.name = 'exp_int_var'
+        self.label_fn = 'Expected Loss'
+        self._integration = integration
+        self._n_samples_imp = n_samples_imp
+        self._iter_imp = iter_imp
+
+        if self._integration == 'importance':
+            self.density_is = RandMaxVar(model=self.model,
+                                         prior=self.prior,
+                                         n_inits=self.n_inits,
+                                         seed=self.seed,
+                                         quantile_eps=self.quantile_eps,
+                                         sampler=sampler,
+                                         n_samples=n_samples,
+                                         sigma_proposals_metropolis=sigma_proposals_metropolis)
+        elif self._integration == 'grid':
+            slices_param = ()
+            for bound in self.model.bounds:
+                slice_param = slice(bound[0], bound[1], d_grid)
+                slices_param = slices_param + (slice_param,)
+            grid_param = list(slices_param)
```

Comment:
`grid_param = [slice(b[0], b[1], d_grid) for b in self.model.bounds]`

Some say list comprehensions are more pythonic :)

Reply:
Changed as suggested.